File: AmaroTest/app/src/main/java/com/test/amaro/amarotest/dagger/modules/RepositoryModule.java
package com.test.amaro.amarotest.dagger.modules;

import android.support.annotation.NonNull;
import com.test.amaro.amarotest.data.ProductService;
import com.test.amaro.amarotest.data.repository.ProductRepository;
import com.test.amaro.amarotest.data.repository.ProductRepositoryImpl;
import dagger.Module;
import dagger.Provides;
import javax.inject.Singleton;

@Module
public final class RepositoryModule {

    @NonNull
    @Provides
    @Singleton
    public final ProductRepository provideProductRepository(ProductService service) {
        return new ProductRepositoryImpl(service);
    }
}

File: AmaroTest/app/src/main/java/com/test/amaro/amarotest/dagger/modules/RepositoryModule.kt
package com.test.amaro.amarotest.dagger.modules

import com.test.amaro.amarotest.data.ProductService
import com.test.amaro.amarotest.data.repository.ProductRepository
import com.test.amaro.amarotest.data.repository.ProductRepositoryImpl
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class RepositoryModule {

    @Provides
    @Singleton
    fun provideProductRepository(service: ProductService): ProductRepository {
        return ProductRepositoryImpl(service)
    }
}